modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2171,12 +2171,44 @@ def map(
         PandasDataframe
             A new dataframe.
         """
-        map_fn = (
-            self._partition_mgr_cls.lazy_map_partitions
-            if lazy
-            else self._partition_mgr_cls.map_partitions
-        )
-        new_partitions = map_fn(self._partitions, func, func_args, func_kwargs)
+        if self.num_parts <= 1.5 * CpuCount.get():
+            map_fn = (
+                self._partition_mgr_cls.lazy_map_partitions
+                if lazy
+                else self._partition_mgr_cls.map_partitions
+            )
+            # old way
+            new_partitions = map_fn(self._partitions, func, func_args, func_kwargs)
+        else:
+            axis = (
+                1
+                if abs(self._partitions.shape[0] - CpuCount.get())
+                < abs(self._partitions.shape[1] - CpuCount.get())
+                else 0
+            )
+            column_splits = (
+                self._partitions.shape[0]
+                // (CpuCount.get() // self._partitions.shape[1])
```

Comment:
Can you explain why the number of rows are divided by columns and cpus, I was not able to follow the logic?

Reply:
In this approach we use columnar partitioning, but in some cases we have too few columnar partitions and use very few remote tasks. (for example, if we only have 1 columnar partition, we will only run 1 remote task). To avoid this, we try to split each columnar partition into several parts. Also, we expect that the number of remote tasks should be less than `CpuCount`. 
`self._partitions.shape[1]` is the number of columnar partitions, so we have to split each of them into that many parts: `CpuCount.get() // self._partitions.shape[1]`. 
Then we need to calculate how many row partitions should be included in each virtual partition, so the number of row partitions is divided by the number of splitting.
![image](https://github.com/modin-project/modin/assets/36244538/a653825f-ec1d-4357-989e-3e3fd975a377)